open_optical_gating/cli/cli.py

```diff
@@ -33,22 +39,33 @@
 # TODO create a time-stamped log somewhere
 
 
-class OpticalGater(picamera.array.PiYUVAnalysis):
+class OpticalGater(PiYUVAnalysis):
     """Custom class to convert and analyse Y (luma) channel of each YUV frame.
-    Extends the picamera.array.PiYUVAnalysis class, which has a stub method called analze that is overidden here.
+    Extends the picamera.array.PiYUVAnalysis class, which has a stub method called analyze that is overidden here.
+
+    # TODO: JT writes: I think these should be short strings instead of numbers, to be suitably descriptive of what the mode is, rather than just magic numbers:
+    self.state has several modes:
+       0 - run prospective gating mode (phase locked triggering)
+       1 - re-initialise (clears for mode 2)
+       2 - get period mode (requires user input)
+       3 - adaptive mode (update period but maintain phase lock)
+
     """
 
-    def __init__(self, camera=None, settings=None, ref_frames=None):
+    def __init__(self, camera=None, settings=None, ref_frames=None, ref_frame_period=None):
         """Function inputs:
             camera - the raspberry picam PiCamera object
             settings - a dictionary of settings (see default_settings.json)
         """
 
         # store the whole settings dict
-        # we occassionally store some of this informatione elsewhere too
+        # we occasionally store some of this information elsewhere too
         # that's not ideal but works for now
         self.settings = settings
         # NOTE: there is also self.pog_settings, be careful of this
+        # TODO: JT writes: UGH! Who has responsibility for the settings object? What is the distinction between settings and pog_settings?

```

Comment:
The distinction is blurry and came about through the fact that there were multiple minds working on this project at different times... it seemed fine last time I did anything with these... but now you come to mention it...... this is kinda ridiculous.

Reply:
Basically, pog_settings are my old settings dictionary when I wrote the pog codes three years ago. Settings was a grouping of Patrick or Alex's settings that they were using - some were new to the system, e.g. pi specific, some were from my settings (but they weren't using everything from my settings) and thus... a  new monster was born.